(Re-)Implemented Light3D's property "shadow_reverse_cull_face"

markusneg opened the pull request:
The parameter "shadow_reverse_cull_face" is now passed to the shadow pass so that the mesh back-faces can be used for shadow map calculation. Fixes #50005.

While in 3.x the "shadow_bias" had to be negative for reversed cull faces, in 4.x it seems to work well with positive values. In contrast, the property "shadow_normal_bias" seems not to be normalized as it makes shadow acne on the back-faces worse.

clayjohn commented on servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp:
```diff
@@ -2364,8 +2365,7 @@ void RenderForwardClustered::_render_shadow_append(RID p_framebuffer, const Page
 	{
-		//regular forward for now
-		bool flip_cull = p_use_dp_flip;
+		bool flip_cull = p_use_dp_flip || p_reverse_cull_face;
```
I think this will break the logic of p_use_dp_flip as the value of flip_cull gets overriden. In the case that p_reverse_cull_face is true, p_use_dp_flip is essentially ignored.

clayjohn replied:
Make sure to test with an OmniLight using the "Dual Paraboloid" shadow mode

markusneg committed:
(Re-)Implemented Light3D's property "shadow_reverse_cull_face"

The parameter shadow_reverse_cull_face is now passed to the shadow pass so that the mesh back-faces are used for shadow map calculation.

clayjohn reviewed (approved):
Looks good to me now!
